src/core/MIGRATION.md

```diff
@@ -106,12 +105,10 @@ export class Plugin {
 **[4] `server/index.ts`** is the entry-point into the server-side code of this plugin. It is identical in almost every way to the client-side entry-point:
 
 ```ts
-import { PluginInitializerContext } from '../../../core/server';
+import { PluginInitializer } from '../../../core/server';
 import { Plugin } from './plugin';
 
-export function plugin(initializerContext: PluginInitializerContext) {
-  return new Plugin(initializerContext);
-}
+export const plugin: PluginInitializer<...> = (initializerContext) => new Plugin(initializerContext)
```

Comment:
It's not obvious from the Type what the generic arguments are, so it might be useful to add at least the setup and start generic arguments to your example.

Optional: we should also consider documenting these generics with @typeParameter:
```
/**
 * The `plugin` export at the root of a plugin's `public` directory should conform
 * to this interface.
 *
 * @typeParam TSetup - Plugin's Setup contract
 * @public
 */
export type PluginInitializer<
```